File: JM/Hash/4. Clothes.java
//의상
import java.util.*;

class Solution {
    public int[] cnum = new int[30];
    
    public int solution(String[][] clothes) {
        Map<String, Integer> cmap = new HashMap<String, Integer>();
        Arrays.fill(cnum, 1);
        for(String[] cloth : clothes){
            String cloth_sort = cloth[1];
            if(!cmap.containsKey(cloth_sort))
                cmap.put(cloth_sort, cmap.size());
            cnum[cmap.get(cloth_sort)]++;
        }
        
        int answer = 1;
        for(int i=0;i<cmap.size();i++){
            answer *= cnum[i];
        }
        return answer-1;
    }
}

File: YJ/01_Hash/4_clothes.py
def solution(clothes):
    clothess = {} #옷 딕셔너리

    for item in clothes:
        name = item[0]
        type = item[1]

        if type in clothess:
            clothess[type] += 1 #종류별로 옷 개수 추가
        else:
            clothess[type] = 1
        
    answer = 1 #적어도 하나는 입음

    for i in clothess.values():
        answer *= (i+1) #옷 안입는 경우 포함

    return answer-1 #하나도 안 입은 경우 빼기